docs/madr/decisions/037-configure-all-gateways.md

```diff
@@ -0,0 +1,147 @@
+
+- Users can set things up for all gateways.
+- We can use different defaults for gateways than for the whole Mesh.
+- More explicit
+
+### Negative Consequences
+
+- The name might be a bit confusing.
+
+## Add label to all gateways and target them by kind `MeshSubset`
+
+We could label all gateways with `kuma.io/gateway` label. Currently, we are using this label only on Kubernetes for gateways with value `enabled` for builtin and `provided` for provided gateways. We could start labeling all gateways with it. That would allow us to create a policy with the kind `MeshSubset` and tag selector matching this label.
+
+```yaml
+targetRef:
+  kind: MeshSubset
+  tags:
+    kuma.io/gateway: "enabled"
+```
+
+In this case, we need to propagate the label from the pod to the dataplane object.
+
+### Positive Consequences
+
+- Users can set things up for all gateways.
+- We can use different defaults for gateways than for the whole Mesh.
+- No need for a new Kind.
+
+### Negative Consequences
+
+- Less explicit
+- We need to add the tag to all gateways.
+
+### Remove requirement of `name` when using kind `MeshGateway`
+
+We could remove requirement of the name when using `kind: MeshGateway` and when the name is missing policy targets all gateways.
+
+### Positive Consequences
+
+- Users can set things up for all gateways.
+- No need for a new Kind.
+- Not confusing names
+
+### Negative Consequences
+
+- Users can impact the configuration of all gateways by mistake
+- Change of the current api
+- Problem with ordering when there is more specific policy with the same kind but with name defined
+
+## Use internal tag `gateways: include/exclude/only` when using with `MeshSubset
+
+We can implement a code change that checks if a policy has the tag `gateway`. If the tag is present, we evaluate one of three possible values: `include`, `exclude`, or `only`. If the value of the tag doesn't match any of these possibilities, we use the actual tag value. If the tag is not defined, we consider it as the default value `include`.
+
+```yaml
+targetRef:
+  kind: MeshSubset
+  tags:
+    gateways: "only"
+```
+
+### Positive Consequences
+
+- Users can set things up for all gateways.
+- No need for a new Kind.
+- Not confusing names
+
+### Negative Consequences
+
+- Configuration is not so obvious
+- What about merging `MeshSubset`'s
+- Can break current behavior if the user has already defined the tag
+
+## Add a new field to targetRef `proxyTypes: [Sidecar, Gateway]`
+
+We can add a new field `proxyTypes` to targetRef which allows to apply policy to: 
+* only gateways
+* all dataplanes including gateways
+* all dataplanes excluding gateways
+
+```yaml
+targetRef:
+  kind: Mesh|MeshSubset
+  proxyTypes: ["Sidecar", "Gateway"]
+```
+When field is not defined or empty we assume that the user wants to apply configuration to all sidecar, not gateways. 
+In this situtation we won't break current behaviour. Also, worth mentioning that `proxyTypes` can only be set when 
```

Comment:
Hmm, but today `kind: Mesh` means both gateways and sidecars, no?

Reply:
Yes, I think I forgot to update it. Let me fix